### DISCUSSION TITLE
Data binding concepts

### DISCUSSION THREAD

**Discussion by beppo-ivel** [appcelerator/titanium_mobile#13013] [Q&A]
My index.xml contains a label. The text of this label should be initialized by the last record of my table. Alloy and Backbone offers to use fetch to query the table. Fetch is async and if I want to do some further calculations on the result I have to deal with callbacks or come up with my own promise. I can see the advantage to use alloy/backbone by defining my model and connect it with the view, but it looks like a disadvantage if you want to query data and do further operations on the result. What I want to know is, which pattern is recommended using the async fetch/promise or should I use titanium db-connect where I can execute queries in a sync way (https://docs.axway.com/bundle/Titanium_SDK_allOS_en/page/working_with_a_sqlite_database.html)? Or can I combine both, defining the model via alloy and query it with a titanium db-connector.

index.js
```
(function() {
   Alloy.Models.test.fetch({query: 'SELECT * FROM test ORDER BY id DESC LIMIT 1' ,
                                         success: function(collection, response, options){
                                                Alloy.Models.test.set(response);          				             		   
                                                //do further operations on testString
          						 },
          		 		error: function(collection, response, options){
                                                         //throw error
                                                         }
                                        });
})();
function cleanup() {
    $.destroy();
}
```

index.xml
```
<Alloy>
  <Model src="test" />
    <Window id="window1" title="Window1" onClose="cleanup">
      <Label id="myLbl" text="{test.testString}" />
    </Window>
  </Alloy>
```
models/test.js
```
exports.definition = {
    config: {
        "columns": {
            "testString": "TEXT"
            "id": "INTEGER PRIMARY KEY AUTOINCREMENT"
        },
        "adapter": {
            "type": "sql",
            "collection_name": "test",
            "idAttribute": "id"
        }
     }
}
```

**Comment by m1ga** [comment]
https://wiki.appcelerator.org/display/guides2/Alloy+Data+Binding#AlloyDataBinding-Backbonebinding
You can listen to events on the Collection (e.g. when a model is added or title is changed). Since it is using Backbone you can look at https://backbonejs.org/#Events or search for Backbone examples too. 
Guess that is what you are looking for. Of course for a simple Label I would just change it by hand when you change your model.
